src/sensitivities/array.jl

```diff
@@ -38,3 +38,8 @@ function Nabla.∇(
     u = l + size(A[i], 1)
     return copy(selectdim(ȳ, 1, (l+1):u))
 end
+
+# XXX: Using a `Union` here is invalid for some reason, so do the signatures separately
```

Comment:
Do you need both signatures? Shouldn't one be calling the other?

Reply:
Specifically, I believe the first (Vararg) one calls the secong (Tuple) one.

Reply:
Yeah, that's what Base does, but it doesn't work to do it for Nabla:
```
julia> @test check_errs(x->fill(x, (4, 4)), randn(4, 4), randn(), randn())
Error During Test at REPL[8]:1
  Test threw exception
  Expression: check_errs((x->begin
            fill(x, (4, 4))
        end), randn(4, 4), randn(), randn())
  MethodError: no method matching ∇(::Array{Leaf{Float64},2}, ::Array{Float64,2})
  Closest candidates are:
    ∇(::Any, ::Any, !Matched::Type{Arg{N}}, !Matched::Any...) where N at /home/alex/.julia/dev/Nabla/src/core.jl:182
    ∇(!Matched::Node, ::Any) at /home/alex/.julia/dev/Nabla/src/core.jl:179
    ∇(::Any, !Matched::typeof(getindex), !Matched::Type{Arg{1}}, !Matched::Any, !Matched::AbstractArray, !Matched::AbstractArray, !Matched::Any, !Matched::Any...) at /home/alex/.julia/dev/Nabla/src/sensitivities/indexing.jl:14
    ...
  Stacktrace:
   [1] compute_Dv(::Function, ::Array{Float64,2}, ::Tuple{Float64}, ::Tuple{Float64}) at /home/alex/.julia/dev/Nabla/src/finite_differencing.jl:51
   [2] compute_Dv at /home/alex/.julia/dev/Nabla/src/finite_differencing.jl:54 [inlined]
   [3] check_errs(::Function, ::Array{Float64,2}, ::Float64, ::Float64, ::Float64, ::Float64) at /home/alex/.julia/dev/Nabla/src/finite_differencing.jl:114 (repeats 2 times)
   [4] top-level scope at none:0
   [5] eval(::Module, ::Any) at ./boot.jl:328
   [6] eval_user_input(::Any, ::REPL.REPLBackend) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:85
   [7] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:117 [inlined]
   [8] (::getfield(REPL, Symbol("##26#27")){REPL.REPLBackend})() at ./task.jl:259
ERROR: There was an error during testing
```
That error is not really helpful or informative in any way, but it does work if you explicitly include the `Tuple`-based method.

Reply:
I meant you would do the tuple-based method and not the vararg method

Reply:
Oh, derp.

Reply:
Yep, you're absolutely right, that works as expected. Fixed.